### ISSUE TITLE
Unable to open campaign file

### ISSUE THREAD

**Issue by kipstafoo** [MegaMek/mekhq#1180]
After shutting down MHQ, I am unable to reload my campaign files. The error in the log is below.

I have attached the campaign file plus all custom mechfiles for the campaign.
[bug data.zip](https://github.com/MegaMek/mekhq/files/3082227/bug.data.zip)

`
Attempted to change ammo type of non-ammo
Attempted to change ammo type of non-ammo
java.lang.ClassCastException: megamek.common.weapons.ppc.ISERPPC cannot be cast to megamek.common.AmmoType
	at mekhq.campaign.parts.equipment.LargeCraftAmmoBin.getFullShots(LargeCraftAmmoBin.java:156)
	at mekhq.campaign.parts.equipment.LargeCraftAmmoBin.updateConditionFromEntity(LargeCraftAmmoBin.java:283)
	at mekhq.campaign.unit.Unit.runDiagnostic(Unit.java:584)
	at mekhq.campaign.io.CampaignXmlParser.parse(CampaignXmlParser.java:672)
	at mekhq.campaign.CampaignFactory.createCampaign(CampaignFactory.java:88)
	at mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:210)
	at mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:129)
	at javax.swing.SwingWorker$1.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at javax.swing.SwingWorker.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
`

The offending unit is the Overlord (2762b). By removing this unit and updating anyone assigned to it, the campaign file loads just fine. I tried a smaller test-case by creating a new campaign with only this unit but the campaign file loaded and the unit entries in the two campaign files looked the same so I'm not able to currently offer more insight into why the main campaign file fails to load.

**Comment by kipstafoo** [comment]
As an update, after manually removing the DS, I re-added it and re-assigned all the crew members to it (oddly, all but one had lost their skills so I had to fix that). After doing this I saved and restarted and the new campaign file works. It looks like something got corrupted along the way. I've attached the newly working campaign file in case you want to diff it.


[The Wilde Bunch30590201_7.zip](https://github.com/MegaMek/mekhq/files/3084721/The.Wilde.Bunch30590201_7.zip)

**Comment by sixlettervariables** [comment]
The equipment numbers on that dropship were messed up, was it a refit?
```
11:14:02,711 WARN [mekhq.Utilities] {SwingWorker-pool-1-thread-1} 
Could not unscramble equipment for Overlord (2762b) - WBS Flying Circus (e9d49aed-610a-4272-9d03-36dd4e04cb51)

 - LRM 20 Ammo Bin equipmentNum: 20
 - LRM 20 Ammo Bin equipmentNum: 55

Equipment Parts:
 10: Artemis IV FCS Nose Artemis IV FCS 
 3: ER Large Laser Nose ER Large Laser 
 13: ER PPC Nose ER PPC 
 24: AC/20 Left Side AC/20 
 43: AC/20 Right Side AC/20 
 26: AC/20 Ammo Bin Left Side AC/20 Ammo 
 7: LRM 20 Nose LRM 20 
 20: LRM 20 Ammo Bin Left Side <Incorrect>  (Missing)
 61: LRM 20 Aft LRM 20 
 5: LRM 20 Nose LRM 20 
 55: LRM 20 Ammo Bin Right Side <Incorrect>  (Missing)
 2: Ultra AC/5 Ammo Bin Nose Ultra AC/5 Ammo 
 25: Ultra AC/5 Ammo Bin Left Side Ultra AC/5 Ammo 
 35: Ultra AC/5 Ammo Bin Left Side Ultra AC/5 Ammo 
 44: Ultra AC/5 Ammo Bin Right Side Ultra AC/5 Ammo 
 45: AC/20 Ammo Bin Right Side AC/20 Ammo 
 54: Ultra AC/5 Ammo Bin Right Side Ultra AC/5 Ammo 
 60: Ultra AC/5 Ammo Bin Aft Ultra AC/5 Ammo 
 9: Artemis IV FCS Nose Artemis IV FCS 
 11: ER PPC Nose ER PPC 
 14: Large Pulse Laser Nose Large Pulse Laser 
 16: Medium Pulse Laser Nose Medium Pulse Laser 
 17: Medium Pulse Laser Nose Medium Pulse Laser 
 18: Medium Pulse Laser Nose Medium Pulse Laser 
 19: Medium Pulse Laser Nose Medium Pulse Laser 
 29: Large Pulse Laser Left Side Large Pulse Laser 
 31: Medium Pulse Laser Left Side Medium Pulse Laser 
 32: Medium Pulse Laser Left Side Medium Pulse Laser 
 36: Medium Pulse Laser Left Side Medium Pulse Laser 
 38: Medium Pulse Laser Left Side Medium Pulse Laser 
 48: Large Pulse Laser Right Side Large Pulse Laser 
 50: Medium Pulse Laser Right Side Medium Pulse Laser 
 51: Medium Pulse Laser Right Side Medium Pulse Laser 
 55: Medium Pulse Laser Right Side Medium Pulse Laser 
 57: Medium Pulse Laser Right Side Medium Pulse Laser 
 0: Ultra AC/5 Nose Ultra AC/5 
 20: ER PPC Left Side ER PPC 
 22: Ultra AC/5 Left Side Ultra AC/5 
 27: ER Large Laser Left Side ER Large Laser 
 33: Ultra AC/5 Left Side Ultra AC/5 
 39: ER PPC Right Side ER PPC 
 41: Ultra AC/5 Right Side Ultra AC/5 
 46: ER Large Laser Right Side ER Large Laser 
 52: Ultra AC/5 Right Side Ultra AC/5 
 58: Ultra AC/5 Aft Ultra AC/5 
 64: Artemis IV FCS Aft Artemis IV FCS 
 65: ER PPC Aft ER PPC 
 67: ER PPC Aft ER PPC 

Equipment:
 0: Ultra AC/5 Ultra AC/5
 1: AC Bay AC Bay (Available)
 2: Ultra AC/5 Ammo Ultra AC/5 Ammo
 3: ER Large Laser ER Large Laser
 4: Laser Bay Laser Bay (Available)
 5: LRM 20 LRM 20
 6: LRM Bay LRM Bay (Available)
 7: LRM 20 LRM 20
 8: LRM 20 Artemis-capable Ammo LRM 20 Artemis-capable Ammo (Available)
 9: Artemis IV FCS Artemis IV FCS
 10: Artemis IV FCS Artemis IV FCS
 11: ER PPC ER PPC
 12: PPC Bay PPC Bay (Available)
 13: ER PPC ER PPC
 14: Large Pulse Laser Large Pulse Laser
 15: Pulse Laser Bay Pulse Laser Bay (Available)
 16: Medium Pulse Laser Medium Pulse Laser
 17: Medium Pulse Laser Medium Pulse Laser
 18: Medium Pulse Laser Medium Pulse Laser
 19: Medium Pulse Laser Medium Pulse Laser
 20: ER PPC ER PPC
 21: PPC Bay PPC Bay (Available)
 22: Ultra AC/5 Ultra AC/5
 23: AC Bay AC Bay (Available)
 24: AC/20 AC/20
 25: Ultra AC/5 Ammo Ultra AC/5 Ammo
 26: AC/20 Ammo AC/20 Ammo
 27: ER Large Laser ER Large Laser
 28: Laser Bay Laser Bay (Available)
 29: Large Pulse Laser Large Pulse Laser
 30: Pulse Laser Bay Pulse Laser Bay (Available)
 31: Medium Pulse Laser Medium Pulse Laser
 32: Medium Pulse Laser Medium Pulse Laser
 33: Ultra AC/5 Ultra AC/5
 34: AC Bay AC Bay (Available)
 35: Ultra AC/5 Ammo Ultra AC/5 Ammo
 36: Medium Pulse Laser Medium Pulse Laser
 37: Pulse Laser Bay Pulse Laser Bay (Available)
 38: Medium Pulse Laser Medium Pulse Laser
 39: ER PPC ER PPC
 40: PPC Bay PPC Bay (Available)
 41: Ultra AC/5 Ultra AC/5
 42: AC Bay AC Bay (Available)
 43: AC/20 AC/20
 44: Ultra AC/5 Ammo Ultra AC/5 Ammo
 45: AC/20 Ammo AC/20 Ammo
 46: ER Large Laser ER Large Laser
 47: Laser Bay Laser Bay (Available)
 48: Large Pulse Laser Large Pulse Laser
 49: Pulse Laser Bay Pulse Laser Bay (Available)
 50: Medium Pulse Laser Medium Pulse Laser
 51: Medium Pulse Laser Medium Pulse Laser
 52: Ultra AC/5 Ultra AC/5
 53: AC Bay AC Bay (Available)
 54: Ultra AC/5 Ammo Ultra AC/5 Ammo
 55: Medium Pulse Laser Medium Pulse Laser
 56: Pulse Laser Bay Pulse Laser Bay (Available)
 57: Medium Pulse Laser Medium Pulse Laser
 58: Ultra AC/5 Ultra AC/5
 59: AC Bay AC Bay (Available)
 60: Ultra AC/5 Ammo Ultra AC/5 Ammo
 61: LRM 20 LRM 20
 62: LRM Bay LRM Bay (Available)
 63: LRM 20 Artemis-capable Ammo LRM 20 Artemis-capable Ammo (Available)
 64: Artemis IV FCS Artemis IV FCS
 65: ER PPC ER PPC
 66: PPC Bay PPC Bay (Available)
 67: ER PPC ER PPC
```

**Comment by cordawtryIC** [comment]
Yes, it was a refit from a 3025 Overlord to the 2762b

**Comment by sixlettervariables** [comment]
I'm tentatively closing this as fixed in #1216. This will land in either 0.45.5 or 0.46.
